### PR TITLE
Refactored the code

### LINKED ISSUE #16: .editorconfig?
I was

  wondering

If you could

  please share

your [`.editorconfig`](http://editorconfig.org/) file.

(

Because

  code with too much vertical . . . . . . . . and horizontal spacing

is hard for me

  to sort in my small mind.

)
;)

### PR DESCRIPTION
**Pull request by shahzainb**
1) Fixed code style issues with spacing
2) Added code blocks for if statements
3) Added var definition in for...in loops
4) Added a hasOwnProperty check for objects in for...in loop
5) Renamed options function to setOptions as it classed with this.options object
6) Added .editorconfig file that fixes #16

### CODE REVIEW

File: js/type.js [hunk 1]
```diff
@@ -1,590 +1,537 @@
...
+            a.push(this[i]);
+          }
+        }
+
+        return _slice.call(a, begin, end || a.length);
+      };
+    }
+  },
+
+  setOptions: function (params) {
```
Comment (shahzainb) [inline js/type.js:53]: I renamed this because it classes with the options object.

File: js/type.js [hunk 2]
```diff
@@ -1,590 +1,537 @@
...
+    this.sheets = stylesheets;
+    this.loadStyles(0);
+  },
+
+  runSheets: function () {
+
+    for (var sheet in this.sheetTexts) {
+      for (var property in this.properties) {
+
+        if (this.sheetTexts.hasOwnProperty(sheet) && this.properties.hasOwnProperty(property)) {
```
Comment (shahzainb) [inline js/type.js:137]: When using a for...in loop over objects, you should use hasOwnProperty to get only the keys that are on the object itself and not inherited ones.

**Comment by shahzainb** [comment]
Hi Nathan,

Just cleaned up the code a bit. Since you haven't got tests, you may wish to test this change yourself to make sure it didn't break anything.

Thanks.

File: js/type.js [hunk 3]
```diff
@@ -1,590 +1,537 @@
...
+      while (temp.firstChild) {
+        parent.insertBefore(temp.firstChild, node);
+      }
+
+      parent.removeChild(node);
+    };
+
+    var kernAll = function () {
+
+      for (var i in rules) {
```
Comment (shahzainb) [inline js/type.js:262]: I read in one of the closed issues that browser support is IE9+. I would then suggest moving these for...in loops on the array to Array.forEach.

**Comment by tomByrer** [comment]
:cake:

**Pull request merged**